rpc/executables.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+)
+
+type CasmEntryPoint struct {
+	Offset   *felt.Felt `json:"offset"`
+	Selector *felt.Felt `json:"selector"`
+	Builtins []string   `json:"builtins"`
+}
+
+type EntryPointsByType struct {
+	Constructor []CasmEntryPoint `json:"CONSTRUCTOR"`
+	External    []CasmEntryPoint `json:"EXTERNAL"`
+	L1Handler   []CasmEntryPoint `json:"L1_HANDLER"`
+}
+
+type CasmCompiledContractClass struct {
+	EntryPointsByType EntryPointsByType `json:"entry_points_by_type"`
+	// can't use felt.Felt here because prime is larger than felt
+	Prime                  string          `json:"prime"`
+	CompilerVersion        string          `json:"compiler_version"`
+	Bytecode               []*felt.Felt    `json:"bytecode"`
+	Hints                  json.RawMessage `json:"hints"`
+	BytecodeSegmentLengths []int           `json:"bytecode_segment_lengths,omitempty"`
+}
+
+func (h *Handler) CompiledCasm(classHash *felt.Felt) (*CasmCompiledContractClass, *jsonrpc.Error) {
+	state, stateCloser, err := h.bcReader.HeadState()
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+	defer h.callAndLogErr(stateCloser, "failed to close state reader")
+
+	declaredClass, err := state.Class(classHash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+
+	switch class := declaredClass.Class.(type) {
+	case *core.Cairo0Class:
+		resp, err := adaptCairo0Class(class)
+		if err != nil {
+			return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+		}
+		return resp, nil
+	case *core.Cairo1Class:
+		return adaptCompiledClass(class.Compiled), nil
+	}
+
+	return nil, jsonrpc.Err(jsonrpc.InternalError, "unsupported class type")
+}
+
+func adaptCairo0Class(class *core.Cairo0Class) (*CasmCompiledContractClass, error) {
+	program, err := utils.Gzip64Decode(class.Program)
+	if err != nil {
+		return nil, err
+	}
+
+	var cairo0 zero.ZeroProgram
+	err = json.Unmarshal(program, &cairo0)
+	if err != nil {
+		return nil, err
+	}
+
+	var bytecode []*felt.Felt
+	for _, str := range cairo0.Data {
+		f, err := new(felt.Felt).SetString(str)
+		if err != nil {
+			return nil, err
+		}
+		bytecode = append(bytecode, f)
+	}
+
+	classHints, err := hintRunnerZero.GetZeroHints(&cairo0)
+	if err != nil {
+		return nil, err
+	}
+
+	var hints [][2]any // slice of 2-element tuples where first value is pc, and second value is slice of hints
+	for pc, hintItems := range utils.OrderMap(classHints) {
+		hints = append(hints, [2]any{pc, hintItems})
+	}
+	rawHints, err := json.Marshal(hints)
+	if err != nil {
+		return nil, err
+	}
+
+	adaptEntryPoint := func(ep core.EntryPoint) CasmEntryPoint {
+		return CasmEntryPoint{
+			Offset:   ep.Offset,
+			Selector: ep.Selector,
+			Builtins: nil,
+		}
+	}
+
+	result := &CasmCompiledContractClass{
+		EntryPointsByType: EntryPointsByType{
+			Constructor: utils.Map(class.Constructors, adaptEntryPoint),
```

Comment:
`utils.Map()` - wow, nice. I haven't noticed we have this

Reply:
yep, it's used across all of our codebase in adapters